Limit number of participants in group chat to 10

cammellos opened the pull request:
A few things:

- Limit the number of participants to 10 in group chats.
- Add invites, so that users will only receive messages if accepted the invitation.- 
- Move group chats to a separate topic, falling back to discovery for older clients.
- Handle multi-envelopes confirmations

### Testing

When creating a new group chats, users will be presented with a Join chat button, and won't be receiving messages until the click on Join chat.

If running a previous version (0.9.32) group chats will work just the same (there's no option to join, but they will receive messages).

On upgrade from 0.9.32 existing group chats will have to be "joined" before receiving messages, giving the stage of development of group chats (experimental), I feel this is acceptable, but  I could fix otherwise.

status: ready

cammellos committed:
Limit number of participants in group chat to 10

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 051ab090 | [#1](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/1/) | 2018-12-17 11:34:51 | ~13 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-112100-051ab0-pr.dmg) |
| :heavy_check_mark: | 051ab090 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/1/) | 2018-12-17 11:36:30 | ~15 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181217-112058-051ab0-e2e.app.zip) |
| :heavy_check_mark: | 051ab090 | [#1](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/1/) | 2018-12-17 11:38:30 | ~17 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-112101-051ab0-pr.exe) |
| :heavy_check_mark: | 051ab090 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/1/) | 2018-12-17 11:38:54 | ~17 min | `ios` | [:package: package](https://i.diawi.com/HFkBVt) |
| :heavy_check_mark: | 051ab090 | [#1](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/1/) | 2018-12-17 11:40:31 | ~19 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-112100-051ab0-pr.AppImage) |
| :heavy_check_mark: | 051ab090 | [#1](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/1/) | 2018-12-17 11:41:15 | ~20 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181217-112056-051ab0-e2e.apk) |
| :heavy_check_mark: | 051ab090 | [#1](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/1/) | 2018-12-17 11:44:39 | ~23 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-112057-051ab0-pr.apk) |
| | | | | | | |
| :heavy_check_mark: | cd180527 | [#3](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/3/) | 2018-12-17 16:03:46 | ~14 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-154936-cd1805-pr.dmg) |
| :heavy_check_mark: | cd180527 | [#3](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/3/) | 2018-12-17 16:09:15 | ~19 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-154936-cd1805-pr.AppImage) |
| :heavy_check_mark: | cd180527 | [#3](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/3/) | 2018-12-17 16:09:46 | ~20 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-154938-cd1805-pr.exe) |
| | | | | | | |
| :heavy_check_mark: | cf6e8df9 | [#4](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/4/) | 2018-12-17 18:37:52 | ~12 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-182502-cf6e8d-pr.dmg) |
| :heavy_check_mark: | cf6e8df9 | [#4](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/4/) | 2018-12-17 18:39:22 | ~14 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181217-182457-cf6e8d-e2e.apk) |
| :heavy_check_mark: | cf6e8df9 | [#4](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/4/) | 2018-12-17 18:39:44 | ~14 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181217-182459-cf6e8d-e2e.app.zip) |
| :heavy_check_mark: | cf6e8df9 | [#4](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/4/) | 2018-12-17 18:41:08 | ~16 min | `ios` | [:package: package](https://i.diawi.com/v63ACg) |
| :heavy_check_mark: | cf6e8df9 | [#4](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/4/) | 2018-12-17 18:41:49 | ~16 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-182503-cf6e8d-pr.exe) |
| :heavy_check_mark: | cf6e8df9 | [#4](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/4/) | 2018-12-17 18:43:42 | ~18 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-182458-cf6e8d-pr.apk) |
| :heavy_check_mark: | cf6e8df9 | [#4](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/4/) | 2018-12-17 18:44:16 | ~19 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-182501-cf6e8d-pr.AppImage) |
| | | | | | | |
| :heavy_check_mark: | 9a6625b1 | [#5](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/5/) | 2018-12-18 14:50:39 | ~17 min | `ios` | [:package: package](https://i.diawi.com/TChhSh) |
| :heavy_check_mark: | 9a6625b1 | [#5](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/5/) | 2018-12-18 14:50:43 | ~17 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181218-143337-9a6625-e2e.app.zip) |
| :heavy_check_mark: | 9a6625b1 | [#5](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/5/) | 2018-12-18 14:52:08 | ~18 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181218-143340-9a6625-pr.dmg) |
| :heavy_check_mark: | 9a6625b1 | [#5](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/5/) | 2018-12-18 14:55:30 | ~21 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181218-143336-9a6625-pr.apk) |
| :heavy_check_mark: | 9a6625b1 | [#5](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/5/) | 2018-12-18 14:55:35 | ~21 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181218-143339-9a6625-pr.AppImage) |
| :heavy_check_mark: | 9a6625b1 | [#5](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/5/) | 2018-12-18 14:55:41 | ~22 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181218-143335-9a6625-e2e.apk) |
| :heavy_check_mark: | 9a6625b1 | [#5](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/5/) | 2018-12-18 14:55:48 | ~22 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181218-143341-9a6625-pr.exe) |
| | | | | | | |
| :heavy_check_mark: | 7df2b0db | [#6](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/6/) | 2018-12-19 21:31:38 | ~11 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181219-212009-7df2b0-pr.dmg) |
| :heavy_check_mark: | 7df2b0db | [#6](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/6/) | 2018-12-19 21:33:28 | ~13 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181219-212005-7df2b0-pr.apk) |
| :heavy_check_mark: | 7df2b0db | [#6](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/6/) | 2018-12-19 21:34:25 | ~14 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181219-212003-7df2b0-e2e.apk) |
| :heavy_check_mark: | 7df2b0db | [#6](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/6/) | 2018-12-19 21:35:02 | ~14 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181219-212005-7df2b0-e2e.app.zip) |
| :heavy_check_mark: | 7df2b0db | [#6](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/6/) | 2018-12-19 21:35:20 | ~15 min | `ios` | [:package: package](https://i.diawi.com/ezX4NH) |
| | | | | | | |
| :heavy_check_mark: | c1496e0a | [#7](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/7/) | 2018-12-24 13:53:23 | ~13 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181224-134000-c1496e-pr.apk) |
| :heavy_check_mark: | c1496e0a | [#7](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/7/) | 2018-12-24 14:00:59 | ~20 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181224-133959-c1496e-e2e.apk) |
| :heavy_check_mark: | c1496e0a | [#7](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/7/) | 2018-12-24 14:04:16 | ~24 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181224-134003-c1496e-pr.dmg) |
| :heavy_check_mark: | c1496e0a | [#7](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/7/) | 2018-12-24 14:04:58 | ~24 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181224-134001-c1496e-e2e.app.zip) |
| :heavy_check_mark: | c1496e0a | [#7](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/7/) | 2018-12-24 14:05:12 | ~25 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181224-134004-c1496e-pr.exe) |
| :heavy_check_mark: | c1496e0a | [#7](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/7/) | 2018-12-24 14:06:17 | ~26 min | `ios` | [:package: package](https://i.diawi.com/yBSeZV) |
| :heavy_check_mark: | c1496e0a | [#7](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/7/) | 2018-12-24 14:06:45 | ~26 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181224-134002-c1496e-pr.AppImage) |
| | | | | | | |
| :x: | c9cdb62b | [#8](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/8/) | 2018-12-27 09:24:17 | ~52 sec | `ios-e2e` | [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/8/consoleText) |
| :heavy_check_mark: | c9cdb62b | [#8](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/8/) | 2018-12-27 09:35:45 | ~12 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181227-092327-c9cdb6-pr.dmg) |
| :heavy_check_mark: | c9cdb62b | [#8](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/8/) | 2018-12-27 09:36:55 | ~13 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181227-092323-c9cdb6-e2e.apk) |
| :heavy_check_mark: | c9cdb62b | [#8](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/8/) | 2018-12-27 09:37:31 | ~14 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181227-092324-c9cdb6-pr.apk) |
| :heavy_check_mark: | c9cdb62b | [#8](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/8/) | 2018-12-27 09:39:54 | ~16 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181227-092328-c9cdb6-pr.exe) |
| :heavy_check_mark: | c9cdb62b | [#8](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/8/) | 2018-12-27 09:41:50 | ~18 min | `ios` | [:package: package](https://i.diawi.com/trnPGR) |
| :heavy_check_mark: | c9cdb62b | [#8](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/8/) | 2018-12-27 09:44:19 | ~20 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181227-092326-c9cdb6-pr.AppImage) |
| | | | | | | |
| :heavy_check_mark: | c6f449cb | [#9](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/9/) | 2018-12-28 11:02:49 | ~21 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181228-104104-c6f449-pr.apk) |
| :x: | c6f449cb | [#9](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/9/) | 2018-12-28 11:24:20 | ~1 min 18 sec | `windows` | [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/9/consoleText) |
| :heavy_check_mark: | c6f449cb | [#9](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/9/) | 2018-12-28 11:42:14 | ~18 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181228-112338-c6f449-pr.AppImage) |
| :heavy_check_mark: | c6f449cb | [#9](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/9/) | 2019-01-07 07:51:38 | ~13 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190107-073741-c6f449-e2e.apk) |
| | | | | | | |
| :heavy_check_mark: | 06bcda04 | [#9](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/9/) | 2019-01-07 08:02:14 | ~15 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190107-074651-06bcda-e2e.app.zip) |
| :heavy_check_mark: | 06bcda04 | [#1](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/1/) | 2019-01-07 08:02:28 | ~15 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190107-074654-06bcda-pr.dmg) |
| :heavy_check_mark: | 06bcda04 | [#10](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/10/) | 2019-01-07 08:03:53 | ~16 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190107-074654-06bcda-pr.exe) |
| :heavy_check_mark: | 06bcda04 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/1/) | 2019-01-07 08:05:01 | ~18 min | `ios` | [:package: package](https://i.diawi.com/obiG32) |
| | | | | | | |
| :heavy_check_mark: | fd1016b2 | [#10](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/10/) | 2019-01-07 08:07:38 | ~15 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190107-075142-fd1016-e2e.apk) |
| | | | | | | |
| :heavy_check_mark: | 06bcda04 | [#10](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/10/) | 2019-01-07 08:11:39 | ~24 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190107-074653-06bcda-pr.AppImage) |
| | | | | | | |
| :heavy_check_mark: | fd1016b2 | [#10](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/10/) | 2019-01-07 08:17:38 | ~15 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190107-080228-fd1016-e2e.app.zip) |
| :heavy_check_mark: | fd1016b2 | [#2](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/2/) | 2019-01-07 08:17:44 | ~15 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190107-080239-fd1016-pr.dmg) |
| :heavy_check_mark: | fd1016b2 | [#11](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/11/) | 2019-01-07 08:20:45 | ~16 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190107-080403-fd1016-pr.exe) |
| :heavy_check_mark: | fd1016b2 | [#2](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/2/) | 2019-01-07 08:22:20 | ~17 min | `ios` | [:package: package](https://i.diawi.com/Ua3pPY) |
| :heavy_check_mark: | fd1016b2 | [#11](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/11/) | 2019-01-07 08:28:18 | ~16 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190107-081149-fd1016-pr.AppImage) |
| :heavy_check_mark: | fd1016b2 | [#10](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/10/) | 2019-01-07 12:53:07 | ~14 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190107-123811-fd1016-pr.apk) |
| | | | | | | |
| :heavy_check_mark: | 4a880e18 | [#11](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7112/11/) | 2019-01-08 07:43:40 | ~13 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190108-072952-4a880e-e2e.app.zip) |
| :heavy_check_mark: | 4a880e18 | [#3](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7112/3/) | 2019-01-08 07:44:30 | ~14 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190108-072955-4a880e-pr.dmg) |
| :heavy_check_mark: | 4a880e18 | [#11](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7112/11/) | 2019-01-08 07:44:56 | ~15 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190108-072951-4a880e-e2e.apk) |
| :heavy_check_mark: | 4a880e18 | [#11](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7112/11/) | 2019-01-08 07:45:15 | ~15 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190108-072952-4a880e-pr.apk) |
| :heavy_check_mark: | 4a880e18 | [#3](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7112/3/) | 2019-01-08 07:47:15 | ~17 min | `ios` | [:package: package](https://i.diawi.com/SkExKj) |
| :heavy_check_mark: | 4a880e18 | [#12](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7112/12/) | 2019-01-08 07:47:35 | ~17 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190108-072954-4a880e-pr.AppImage) |
| :heavy_check_mark: | 4a880e18 | [#12](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7112/12/) | 2019-01-08 07:47:41 | ~17 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190108-072956-4a880e-pr.exe) |

cammellos committed:
Add joining of group chats

Members will now have to explicitly join a group chat to start receiving
messages from it.

Messages are still sent to users who have not joined for backward
compatibility.
Group updates are unaffected.

cammellos committed:
Move group chats to their own topic

This commit moves group chats to their own topic, based on the randomly
generated chat-id. It falls back on the discovery topic for those peers
who we can't fingerprint the version, for backward compatibility.

cammellos committed:
Extend message hash confirmation for group chats

Signed-off-by: Andrea Maria Piana <andrea.maria.piana@gmail.com>

mandrigin commented on src/status_im/ui/components/checkbox/view.cljs:
```diff
@@ -3,13 +3,16 @@
+                         :disabled            (and (not checked?)
+                                                   disabled?)
                          :accessibility-label accessibility-label}]]
       [react/touchable-highlight (merge {:style               (merge styles/wrapper style)
```
hmm, shouldn't we add this `disabled?` flag to iOS too?

cammellos replied:
I will have a look actually :)

rasom commented on src/status_im/group_chats/core.cljs:
```diff
@@ -399,6 +465,27 @@
      (map #(assoc % :from from) events))
    all-updates))
 
+(defn get-inviter-pk [my-public-key {:keys [membership-updates] :as chat}]
```
`chat` is not used

rasom commented on src/status_im/transport/message/core.cljs:
```diff
@@ -57,6 +57,19 @@
                                                          :chat    updated-chat})]}))
+(fx/defn check-confirmations [{:keys [db] :as cofx} status chat-id message-id]
+  (when-let [{:keys [pending-confirmations not-sent]} (get-in db [:transport/message-ids->confirmations message-id])]
+    (if (= 0 (dec pending-confirmations))
```
`zero?`

rasom commented on src/status_im/transport/message/core.cljs:
```diff
@@ -57,6 +57,19 @@
      :data-store/tx [(transport-store/save-transport-tx {:chat-id chat-id
                                                          :chat    updated-chat})]}))
+(fx/defn check-confirmations [{:keys [db] :as cofx} status chat-id message-id]
+  (when-let [{:keys [pending-confirmations not-sent]} (get-in db [:transport/message-ids->confirmations message-id])]
```
80 chars limit

asemiankevich commented:
@cammellos for some reason i dont see 7112 branch in https://ci.status.im/job/status-react/job/prs/job/android/view/change-requests/ (android), could you please check?

<blockquote><img src="/static/59a7919b/favicon.ico" width="48" align="right"><div><strong><a href="https://ci.status.im/job/status-react/job/prs/job/android/view/change-requests/">Pull Requests (17) [status-react » prs » android] [Jenkins]</a></strong></div></blockquote>

cammellos commented:
@asemiankevich should be building now, thanks!

asemiankevich commented:
@errorists maybe you could take a look at this from design point of view? I suggest move this `You can select 9 more participants` to this nice placeholder in the middle of the screen as a minimum^ also apply the fonts and colors that we are using across the app. 

Anyway your input here is highly appreciated!

![image](https://user-images.githubusercontent.com/28752642/50769489-2283c980-1295-11e9-8e20-cd0d70126af6.png)

there are some more screens down here

![image](https://user-images.githubusercontent.com/28752642/50769750-0f252e00-1296-11e9-801a-0defe1b24af4.png)

![image](https://user-images.githubusercontent.com/28752642/50769806-32e87400-1296-11e9-9666-56bf5cb5bfaf.png)

errorists commented:
@asemiankevich sure, I wasn't aware we're limiting the max number of users but on the design side I'd like us to follow WhatsApp here with a design like this 

<img width="363" alt="screenshot 2019-01-07 at 14 13 45" src="https://user-images.githubusercontent.com/4932774/50770274-fb260000-1286-11e9-955d-47ce44213341.png">

The designs are in [this Figma](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=231%3A0)

cammellos commented:
@asemiankevich @errorists thanks, I am happy to implement it, but in a separate PR if that's ok

asemiankevich commented:
> @asemiankevich @errorists thanks, I am happy to implement it, but in a separate PR if that's ok

@cammellos absolutely fine! thanks Andrea

asemiankevich commented:
1. Designs will be implemented later in separate PR. Designs are [here](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=231%3A0)
2. Questions I have so far:
- no option to decline the invite. Only Join button. @errorists maybe we can think about something here? Not needed? Any user action if I dont want to join this group i was invited to?
- no group chat shown if user was re-invited to the chat after he has deleted it on his device. Steps:

- [ ] create group chat by device A (admin) with device B (non-admin)
- [ ] as device B accept the invite
- [ ] as device B delete the chat. Chat is removed from Home screen. System message in a group chat states that user B left the group (can be checked on device A).
- [ ] as device A re-add user B to the same group chat
- [ ] as device B check that nothing appears on home screen (no group chat shown). **This is same as in develop.**
- [ ] as device A send some messages to this group chat
- [ ] as device B check that you still dont see anything. **This seems to be wrong and in develop user B will see group chat and messages again.**

- some warning shown when i am on Home screen, not sure if this is related to this PR though. 
![image](https://user-images.githubusercontent.com/28752642/50771400-9cb74c80-129b-11e9-880b-8ee89dd32258.png)
- as a user that was invited to the group chat i am already listed as a member even i did not touch the  `Join group` button. Also, admin could make me admin and then i can add / remove members even i did not join the group yet. Do we want to limit any user functionality prior he taps the `Join` button? @errorists also question to you.

![image](https://user-images.githubusercontent.com/28752642/50771638-3da60780-129c-11e9-9b22-d3be6e42e10a.png)

- as admin of chat i create with release version  i see `Join chat` button when upgraded from released version to this PR. Also some random inviter is indicated (this is a member of the chat though). When i tap Join -> i see this system message about joining.

![image](https://user-images.githubusercontent.com/28752642/50775156-7a76fc00-12a6-11e9-83a3-5a392dadb74f.png)

![image](https://user-images.githubusercontent.com/28752642/50775172-7f3bb000-12a6-11e9-833a-03d6be686156.png)

- not clear what is expected when i upgrade a chat which has more than 10 members? No errors for now. 
- `add members` is not hidden if i upgrade the chat which has more than 10 participants already. I can't add more though. @errorists any input?

cammellos commented:
> no group chat shown if user was re-invited to the chat after he has deleted it on his device

that I will fix

>  some warning shown when i am on Home screen, not sure if this is related to this PR though.

I think was already present, it won't be present on production, but I will take a look at some point

> add members is not hidden if i upgrade the chat which has more than 10 participants already.

I will fix

The rest seems to be mainly ux issues, and if it's ok, I would address in separate prs.

Thanks!

cammellos commented:
@asemiankevich I have addressed the 2 points (warning is still present). Thanks!

asemiankevich commented:
@cammellos thanks checking now! meanwhile what about this ?

> as admin of chat i create with release version i see Join chat button when upgraded from released version to this PR. Also some random inviter is indicated (this is a member of the chat though). When i tap Join -> i see this system message about joining.

Quite confusing to me to see this Join chat with random inviter when i am already a chat creator (after upgrade).

Swader commented:
Is there a specific reason for this limitation? Why a limit at all?

cammellos commented:
@Swader currently performance, group chats use a stronger encryption and it's more expensive as more users join. 
Currently the focus is to make sure it works correctly, if we don't limit the number of users it's hard to understand which problems are due to performance degradation and which are due to the mechanics.
As we get more feedback we can then work on improving performance and increasing the limit (or removing it if it becomes feasible of course).

errorists commented:
And what about the 'Join Chat' gateway, is there a specific reason we're adding this now?

cammellos commented:
@errorists yes, partially based on user feedback (people complained about being added to chats by people that they have not added in cotacts), partially is something we discussed with grame previously (the model was wechat), and also there are some technical reasons (having a confirmation that the user wants to receive messages from this group chats helps with some performance optimization).

Swader commented:
@cammellos I see, thanks. In that case a message clarifying this limit at the invite screen would be useful from a UX perspective (@hesterbruikman), but it needs to be worded correctly. People's train of thought will be: "Wtf, Telegram is encrypted and I can have dozens of thousands of members in a group". Perhaps the message could contain a link which leads to an explanatory blog post.

cammellos commented:
@Swader yes fair point, happy to add something on those lines.
For the specific example, telegram group chats are not end-to-end encrypted (only client-server, if the server is compromised all the conversation can be decrypted), and that's why they can have a large number of users.

the encryption we use is similar to signal's, where the group is unlimited, but performance degradation happens on large groups. What's app uses a similar method and limits to 256 (which is clearly much more than 10, but also we are not whatsapp yet ;) )

Swader commented:
I see. This is all something the average user won't know, and makes perfect content for the blog post explainer.

errorists commented:
@cammellos thanks for explanation. I went ahead and did a design for the 'Join' prompt, I've also added an option to decline the invitation in which case the group chat should be removed from your Home Screen. I'm not sure if we should disclose the number of members inside the group until you've joined, I figure maybe not. [Figma link](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=1741%3A0) Please keep me in the loop if you're doing this and the design for capping max group members as a separate PR

<img width="404" alt="screenshot 2019-01-08 at 11 24 54" src="https://user-images.githubusercontent.com/4932774/50825120-80b5b880-1338-11e9-88c7-2bd29825438d.png">

On the subject of informing user about the decisions we made and limitations set on them. I honestly don't believe anyone would care enough to read through a blog post :) It is a fluid limitation, one that we can stress test, I'm totally fine with not telling it's due to performance limitations of the protocol.

asemiankevich commented:
@errorists thanks !! Exactly what i was asking for!

cammellos commented:
@errorists thanks, it will be in a separate PR, I will keep you in the loop!

Swader commented:
@errorists you're probably right IRT majority of users. I'm just a curious minority :)

errorists commented:
Also to answer @asemiankevich  previous questions. 
I would leave the Add members button visible, but once inside use the same mechanism to inform of the total user number and if it's reached the limitation, exactly the same as when you're creating a new group.
Users should not be visible on the group member list until they've joined and similarly should not be counted against the total users limit.

cammellos commented:
> I would leave the Add members button visible, but once inside use the same mechanism to inform of the total user number and if it's reached the limitation, exactly the same as when you're creating a new group.

I can revert the change if @asemiankevich is ok (possibly in a different pr, if this tests ok soon)

> Users should not be visible on the group member list until they've joined
 
That we can do, although an admin should see all the invitations I take?

>  and similarly should not be counted against the total users limit.

This is tricky, essentially because if I understand correctly, it means I could "over-invite" a chat (send 15 invitations, and only when 10 have joined prevent to send invitations).

If that is the desired behavior it adds quite a few technical issues (I can go into details, but basically it might end up with having a chat over-joined, and having to exclude members, and we need to make sure all the participants have a consistent view of the chat etc), so I'd rather keep it as it is, unless we feel strongly about this point.

errorists commented:
> I can revert the change if @asemiankevich is ok (possibly in a different pr, if this tests ok soon)

different PR along with the change in[ how the total number of users is shown](https://github.com/status-im/status-react/pull/7112#issuecomment-451932992) would be great, together these will make sense :) All of those designs are updated and reflected in the [Group Chat Figma](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=227%3A320)

> That we can do, although an admin should see all the invitations I take?

Good point, yes admin should see all invitations.

> This is tricky, essentially because if I understand correctly, it means I could "over-invite" a chat (send 15 invitations, and only when 10 have joined prevent to send invitations).

I thought it would be much simpler and overlook the details, it can be left as is then

asemiankevich commented:
@cammellos yess the suggestion from @errorists works to me

hesterbruikman commented:
Such rapid progress here! Great work!
